spec/services/catalog/provider_control_parameters_spec.rb

```diff
@@ -17,30 +13,43 @@
     TopologicalInventoryApiClient::ContainerProject.new('name'      => project2_name,
                                                         'source_id' => "2")
   end
-  let(:ti_class) { class_double("TopologicalInventory").as_stubbed_const(:transfer_nested_constants => true) }
 
-  context "#{described_class}#process" do
-    before do
-      allow(ti_class).to receive(:call).and_yield(api_instance)
+  before do
+    allow(ManageIQ::API::Common::Request).to receive(:current_forwardable).and_return(default_headers)
+  end
+
+  around do |example|
+    with_modified_env(:TOPOLOGICAL_INVENTORY_URL => "http://localhost") do
+      example.call
+    end
+  end
 
-      container_projects = double('links' => {}, 'meta' => {}, 'data' => [project2, project1])
-      expect(api_instance).to receive(:list_source_container_projects).and_return(container_projects)
+  describe "#process" do
+    let(:container_project_response) do
+      TopologicalInventoryApiClient::ContainerProjectsCollection.new(:data => [project2, project1])
     end
 
-    let(:namespace_list) do
-      data = provider_control_parameters.process.data
-      data['properties']['namespace']['enum']
+    before do
+      stub_request(:get, "http://localhost/api/topological-inventory/v1.0/sources/1/container_projects")
```

Comment:
haha I did this in #396 as well, but this is better I think. I just hardcoded the array.